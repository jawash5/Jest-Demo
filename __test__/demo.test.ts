import {add} from "../src/test/utils";
import HelloWorld from "../src/components/HelloWorld.vue";
import {mount, shallowMount} from "@vue/test-utils";
import demo from '../src/components/demo';
import App from '../src/App.vue'

describe('demo', () => {
  it('add', () => {
    expect(add(1,2)).toBe(3)
  })

  it('组件测试', function () {
    const wrapper = mount(HelloWorld, {
      props: {
        msg: 'Hello Vue 3 + TypeScript + Vite'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
    expect(wrapper.get("h1").html()).toBe('<h1>Hello Vue 3 + TypeScript + Vite</h1>')
    wrapper.unmount()
  });

  it('jsx测试', () => {
    const wrapper = mount(demo)
    expect(wrapper.get("div").html()).toBe('<div>HelloWorld</div>')
    wrapper.unmount()
  });

  it('图片测试', function () {
    const wrapper = shallowMount(App)
    expect(wrapper.find("img").exists()).toBe(true)
    wrapper.unmount()
  });

  it('E2E测试', async function () {
    const wrapper = mount(HelloWorld, {
      props: {
        msg: 'Hello Vue 3 + TypeScript + Vite'
      }
    })
    await wrapper.get("button").trigger('click')
    expect(wrapper.get("button").html()).toContain('1')
    wrapper.unmount()
  });

  it('图片测试', function () {
    const wrapper = mount(App)
    console.log(wrapper.html())
    expect(wrapper.find("img").exists()).toBe(true)
    wrapper.unmount()
  });

})