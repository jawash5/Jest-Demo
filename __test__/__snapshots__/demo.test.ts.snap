// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`demo 组件测试 1`] = `
"<h1>Hello Vue 3 + TypeScript + Vite</h1>
<p> Recommended IDE setup: <a href=\\"https://code.visualstudio.com/\\" target=\\"_blank\\">VSCode</a> + <a href=\\"https://marketplace.visualstudio.com/items?itemName=octref.vetur\\" target=\\"_blank\\"> Vetur </a> or <a href=\\"https://github.com/johnsoncodehk/volar\\" target=\\"_blank\\">Volar</a> (if using <code>&lt;script setup&gt;</code>) </p>
<p>See <code>README.md</code> for more information.</p>
<p><a href=\\"https://vitejs.dev/guide/features.html\\" target=\\"_blank\\"> Vite Docs </a> | <a href=\\"https://v3.vuejs.org/\\" target=\\"_blank\\">Vue 3 Docs</a></p><button type=\\"button\\">count is: 0</button>
<p> Edit <code>components/HelloWorld.vue</code> to test hot module replacement. </p>"
`;
